include/swift/AST/DiagnosticsSema.def

```diff
@@ -2824,6 +2824,8 @@ ERROR(diff_params_clause_no_inferred_parameters,PointsToFirstBadToken,
       "no differentiation parameters could be inferred; must differentiate "
       "with respect to at least one parameter conforming to 'Differentiable'",
       ())
+ERROR(diff_params_clause_inout_argument,none,
+      "inout types (%0) cannot be differentiated with respect to", (Type))
```

Comment:
Since `inout` types aren't a thing in user-facing Swift, we should talk about `inout` parameters instead. Also, diagnostics often use single quotes around code. 
```suggestion
      "'inout' parameters (%0) cannot be differentiated with respect to", (Type))
```